fairscale/experimental/optim/dynamic_loss_scaler.py

```diff
@@ -0,0 +1,207 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+"""
+To prevent underflow or overflow of gradients, DynamicLossScaler is used to
+dynamically scale up and down gradients by scaling the loss.
+"""
+
+
+from collections import defaultdict
+from enum import Enum
+from typing import Dict, List, Optional
+
+import torch
+
+
+class OptState(Enum):
+    READY = 0
+    UNSCALED = 1
+    STEPPED = 2
+
+
+def _refresh_per_optimizer_state() -> OptState:
+    return OptState.READY
+
+
+class DynamicLossScaler(object):
+    """An instance ``scaler`` helps perform the steps of gradient scaling
+    conveniently.
+
+    * ``scaler.scale(loss)`` multiplies a given loss by ``scaler``'s current scale factor.
```

Comment:
We generally don't specify the methods in the doc string.  The  python doc string usually has the following format:
```
single line comment

multiple line comments

Args:

Returns:

Raises:
```

Reply:
I referred to https://pytorch.org/docs/stable/_modules/torch/cuda/amp/grad_scaler.html#GradScaler as a template. Is there any  wiki or doc that I can follow for this codebase?